### PR TITLE
[pilot] Move log level for Envoy proxy to a flag

### PR DESCRIPTION
**Pull request by nmittler**

### CODE REVIEW

File: pilot/proxy/envoy/watcher.go [hunk 1]
```diff
@@ -202,14 +202,29 @@ type envoy struct {
 	extraArgs []string
 }
 
+// ProxyLogLevel defines the value of the log level flag (-l) passed to the Envoy proxy upon creation.
+type ProxyLogLevel string
+
+const (
+	// ProxyLogLevelNone environment flag
```
Comment (kyessenov) [inline pilot/proxy/envoy/watcher.go]: Maybe we should just pass a string? There is also notice, error, and warning levels, I think.

Comment (nmittler) [reply pilot/proxy/envoy/watcher.go]: done.

**Comment by nmittler** [comment]
@kyessenov PTAL

**Comment by kyessenov** [comment]
I think this is OK to get back to what we had.

The right solution is not to use the flag, but control logging dynamically. Changing agent start-up flags is very awkward in k8s.

**Comment by nmittler** [comment]
> I think this is OK to get back to what we had. The right solution is not to use the flag, but control logging dynamically. Changing agent start-up flags is very awkward in k8s.

@kyessenov I guess you're suggesting that this be somehow incorporated into the CRD then? Or is there some other dynamic mechanism in the system?

**Comment by kyessenov** [comment]
Envoy has dynamic logging capability on admin channel. Admin channel is
plaintext though (!!!). This could be authorized through Pilot and then
controlled centrally Pilot.

**Comment by nmittler** [comment]
@kyessenov It would seem that log level was already static, since it was passed as a startup parameter to Envoy.  So this PR isn't changing existing behavior ... or am I missing something?

**Comment by kyessenov** [comment]
It's good as it is, thanks. I'm just spelling out that the old way was not
adequate :)

**Pull request merged**